src/profile.c

```diff
@@ -83,9 +83,8 @@
 **     we provide -P (profiling) and -c (code coverage) options to the GAP
 **     executable so the user can start before code loading starts
 **
-**  3) Operating at just a line basis can sometimes be too course. We can
-**     see this when we use ActivateProfileColour. However, without some
-**     serious additional overheads, I can't see how to provide this
+**  3) Operating at just a line basis can sometimes be too course. However,
+**     without some serious additional overheads, I can't see how to provide this
 **     functionality in output (basically we would have to store
 **     line and character positions for the start and end of every expression).
```

Comment:
Actually I do think we could provide that without a problem nor overhead, by replacing the line number in the stat by a byte offset into the file. Converting this into line number + offset into the line can then be deferred to a post-processing step.

Reply:
There is a bit more problem here ( I could describe it better).

We tend to set up statements we tend to (as you would expect) parse, then make a `Stat`, so if we just grabbed the position when making the `Stat` we would point to the end of the statement / loop, / whatever.

With just covering lines this (usually) isn't too bad, but if you want to capture expressions usefully you probably want to catch their start and end (as expressions are often nested), and then we'd need to catch the start of expressions.